File: javascript/com/github/devginie/codility/lesson2/OddOccurrencesinArray.js
/**
 * https://codility.com/programmers/task/odd_occurrences_in_array/
 * Find value that occurs in odd number of elements.
 *
 * @author devginie, tinywind
 * score: 100
 * */
function solution(A) {
    var unpaired = {},
    i, max;
    
    for (i = 0, max = A.length; i < max; i += 1) {
        if (unpaired[A[i]]) {
            unpaired[A[i]] = false;
            continue;
        }
        unpaired[A[i]] = true;
    }
    for (var k in unpaired) {
        if (unpaired.hasOwnProperty(k) && unpaired[k]) {
            return Number(k);
        }
    }
    return -1;
}

File: java/org/tinywind/codility/com/lesson2/OddOccurrencesInArray.java
package org.tinywind.codility.com.lesson2;

import java.util.Arrays;
import java.util.TreeSet;

/**
 * https://codility.com/programmers/task/odd_occurrences_in_array/
 * Find value that occurs in odd number of elements.
 *
 * @author http://tinywind.blog.me/220752072231
 */
public class OddOccurrencesInArray {
    public static void main(String[] args) {
        p(2, 1, 2, 3, 3);
        p(10, 10, 1);
        p(10);
    }

    private static void p(int... A) {
        System.out.println("A(" + Arrays.toString(A) + ") : " + solution(A));
    }

    private static int solution(int[] A) {
        TreeSet<Integer> unpaired = new TreeSet<>();
        for (int i : A) {
            if (unpaired.remove(i)) {
                continue;
            }
            unpaired.add(i);
        }
        return unpaired.first();
    }
}
